src/google/protobuf/unittest_import_public.proto

```diff
@@ -35,6 +35,7 @@ syntax = "proto2";
 package protobuf_unittest_import;
 
 option java_package = "com.google.protobuf.test";
+option csharp_namespace = "Google.ProtocolBuffers.TestProtos";
```

Comment:
Yes, I wondered about that. It ends up looking "odd" to have unconventional namespaces in the test code - I don't have much opinion either way though. If we don't want this, I'll create another commit which changes the namespace in the generated (but present-in-git) code.